lib/context.js

```diff
@@ -116,7 +116,12 @@ var proto = module.exports = {
     }
 
     // unset all headers, and set those specified
-    this.res._headers = {};
+    if (this.res.getHeaderNames) {
+      this.res.getHeaderNames().forEach(function(name) { this.removeHeader(name) }, this.res);
```

Comment:
This line gets tested when the tests a are run with node 8 (test/context/onerror.js). I did a manual test run using node master branch.